### ISSUE TITLE
accumulate_grad_batches flag in trainer changes cosine decay scheduler incorrectly

### ISSUE THREAD

**Issue by exnx** [Lightning-AI/lightning#12894]
## 🐛 Bug

When using the `accumulate_grad_batches` flag in trainer, I noticed it changes my scheduler LR differently.  I am using TIMM's cosine decay scheduler.  And when I use 1 gpu, I set `accumulate_grad_batches` = 4, and when I use 4 gpus, I set `accumulate_grad_batches` = 1.  Then I noticed the LR scheduler (in wandb) displayed different LR during training.  For the `accumulate_grad_batches` = 4, the LR decayed faster.

My understand is the schedule should only depend on the epoch number, not the batch size or `accumulate_grad_batches` value.  Did anyone else notice something similar before?

I am using PTL version 1.5, torch 1.11, and TIMM 0.5.4.  I am training videos (HMDB) using pytorchvideo library as well.

Thanks!

cc @borda @rohitgr7

**Comment by rohitgr7** [comment]
> the LR decayed faster.

what do you mean by faster? if you are using the same batch_size in both the cases then based on step, while using 4 GPUs the decay will be faster as compared to the other case since the effective batch size per step is 4 times here but in the first case the update will happen after every 4 steps (accumulation factor).
